### ISSUE TITLE
Cannot read properties of undefined (reading 'getInitialProps')

### ISSUE THREAD

**Issue by silentim** [vercel/next.js#36221]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

next: 12.1.1-12.1.6-canary.2
react: 17.0.2 or 18.0.0
node: 14.19.1
system: windows

### What browser are you using? (if relevant)

chrome 100.0.4896.88

### How are you deploying your application? (if relevant)

next start

### Describe the Bug

![error](https://user-images.githubusercontent.com/65599013/163697340-c6f59655-ff3b-4290-b728-d1ed0a674d2c.png)

### Expected Behavior

no error in console and display page content

### To Reproduce

```
//app.ts

const MyApp = ({ Component, pageProps }: AppProps) => {
    return <Component {...pageProps} />;
};

MyApp.getInitialProps = async (appContext: AppContext) => {
    const some = await fetch('....');

    const appProps = await App.getInitialProps(appContext);
    return { ...appProps, pageProps: { ...appProps.pageProps, some } };
};

export default MyApp;
```

```
//index.ts

export default function Home(){
    return (
        <div>some text</div>
    )
}

export const getStaticProps: GetStaticProps = async () => {
    return { props: { } };
}
```

Then npm run build, everything is ok and built successfully.
Now npm run start, get error like this in console (see image attached) and there is nothing displaying in browser, just blank:

A client-side exception has occurred, see here for more info: https://nextjs.org/docs/messages/client-side-exception-occurred
Error rendering page:  TypeError: Cannot read properties of undefined (reading 'getInitialProps')

No error if next 12.1.0 and more lower version.

**Comment by silentim** [comment]
I found, if some function uses 'zlib' module in app.ts, and swcMinify=true in next.config.js, this error comes like above; If swcMinify=false, page is ok and no error. So then zlib will not  be supported after next 12.1.0?

**Comment by kdy1** [comment]
No, it's a bug of swc minifier. As there are so many minification rules, there are some bugs.
It's a huge project.


<img width="706" alt="image" src="https://user-images.githubusercontent.com/29931815/164410929-f07f3edb-d7d8-457b-ab0b-bc993f3ea265.png">


I'll add this to my private task list

**Comment by C-E-Rios** [comment]
Seeing the same. Setting `swcMinify=true` as recommended by @silentim fixed it.

**Comment by kdy1** [comment]
Is this issue still the case?
As there's no repro it's too hard for me to verify it

**Comment by kdy1** [comment]
@davidhanlonJCM I think so, but I can't verify as there's no input nor repro case.

**Comment by davidhanlonJCM** [comment]
@kdy1 upgrading to next 12.3 appears to have solved the issue for me. In my case the issue appeared when using next 12.1 with `swcMinify:true` in the next.config file. It only showed up when running a full build or deployed to a vercel domain

**Comment by DonikaV** [comment]
in version v13, if I run production (npm run build && npm run start) version I get these errors. if dev - all fine :|

**Comment by kdy1** [comment]
@DonikaV Can you provide a repro?

**Comment by rossignolli** [comment]
i've been getting the same errors in my sentry report.

using **next 12.3.1**

I've already tried to disable swcMinify.
 
i use **getInitialProps** once only in `_document.js `to setup styled-components basead on [with-styled-components](https://github.com/vercel/next.js/blob/canary/examples/with-styled-components/pages/_document.tsx)


```
import Document, { DocumentContext } from 'next/document'
import { ServerStyleSheet } from 'styled-components'

export default class MyDocument extends Document {
  static async getInitialProps(ctx: DocumentContext) {
    const sheet = new ServerStyleSheet()
    const originalRenderPage = ctx.renderPage

    try {
      ctx.renderPage = () =>
        originalRenderPage({
          enhanceApp: (App) => (props) =>
            sheet.collectStyles(<App {...props} />),
        })

      const initialProps = await Document.getInitialProps(ctx)
      return {
        ...initialProps,
        styles: [initialProps.styles, sheet.getStyleElement()],
      }
    } finally {
      sheet.seal()
    }
  }
}
```

My sentry report:

```
TypeError
Cannot read property 'getInitialProps' of undefined

./node_modules/next/dist/shared/lib/utils.js in call at line 66:18

                throw new Error(message);
            }
        }
        // when called from _app `ctx` is nested in `ctx`
        const res = ctx.res || ctx.ctx && ctx.ctx.res;
        if (!App.getInitialProps) {
            if (ctx.ctx && ctx.Component) {
                // @ts-ignore pageProps default
                return {
                    pageProps: yield loadGetInitialProps(ctx.Component, ctx.ctx)
                };




ReferenceError: globalThis is not defined
console
A client-side exception has occurred, see here for more info: 

```

Consequently seems like some users are getting and others not. any ideas?

**Comment by kdy1** [comment]
@rossignolli Xan you try 12 canary?

**Comment by DonikaV** [comment]
> @DonikaV Can you provide a repro?

No, the problem that I don't use anywhere any functions like getInitiallProps. I only use env vars. maybe it because of that, but in version 12* all works fine. In 13 no :|

**Comment by rossignolli** [comment]
> @rossignolli Can you try 12 canary?

 Same :/

**Comment by kdy1** [comment]
Oh if disabling swcMinify does not work it's a different issue

**Comment by DonikaV** [comment]
after I disabled swcMinify works well. But is weird why it is happening :) @kdy1 
I don't use anywhere this function **getInitialProps**...

**Comment by kdy1** [comment]
@DonikaV I can't debug without a repro

**Comment by karolina-szlenk** [comment]
@rossignolli have you already resolved your problem? I faced the same :(

**Comment by rossignolli** [comment]
> 

Analyzing my sentry report I noticed that most of errors come from outdated or unsupported browsers versions.  

But i still trying to figure out a solution.

**Comment by tomgreeEn** [comment]
Also having this issue with 13.0.2

Production Builds using getStaticProps are failing on Vercel are failing with the error:

```
TypeError: Cannot read properties of undefined (reading 'getInitialProps')
--
17:47:51.855 | at Object.41888 (/vercel/path0/.next/server/chunks/2843.js:3024:43)
17:47:51.855 | at __webpack_require__ (/vercel/path0/.next/server/webpack-runtime.js:25:42)
17:47:51.856 | at /vercel/path0/.next/server/chunks/2843.js:3294:94
17:47:51.856 | at Function.__webpack_require__.a (/vercel/path0/.next/server/webpack-runtime.js:89:13)
17:47:51.856 | at Object.34008 (/vercel/path0/.next/server/chunks/2843.js:3259:21)
17:47:51.856 | at __webpack_require__ (/vercel/path0/.next/server/webpack-runtime.js:25:42)
17:47:51.856 | at /vercel/path0/.next/server/chunks/2843.js:3219:93
17:47:51.856 | at Function.__webpack_require__.a (/vercel/path0/.next/server/webpack-runtime.js:89:13)
17:47:51.856 | at Object.6817 (/vercel/path0/.next/server/chunks/2843.js:3205:21)
17:47:51.856 | at __webpack_require__ (/vercel/path0/.next/server/webpack-runtime.js:25:42)
```
...but I can do production builds locally.



swcMinify: false does not fix the error. 

I am also using a custom _document to inject serverside styles, but using Emotion.

My _document file:


```
class MyDocument extends Document {
  render() {
    return (
      <Html>
        {...}
      </Html>
    );
  }
}


MyDocument.getInitialProps = async (ctx) => {
 
  const originalRenderPage = ctx.renderPage;

  const cache = createEmotionCache();
  const { extractCriticalToChunks } = createEmotionServer(cache);

  ctx.renderPage = () =>
    originalRenderPage({
      enhanceApp: (App) =>
        function EnhanceApp(props) {
          return <App emotionCache={cache} {...props} />;
        },
    });

  const initialProps = await Document.getInitialProps(ctx);
  // This is important. It prevents Emotion rendering invalid HTML.
  // See https://github.com/mui/material-ui/issues/26561#issuecomment-855286153

  const emotionStyles = extractCriticalToChunks(initialProps.html);
  const emotionStyleTags = emotionStyles.styles.map((style) => (
    <style
      data-emotion={`${style.key} ${style.ids.join(' ')}`}
      key={style.key}
      // eslint-disable-next-line react/no-danger
      dangerouslySetInnerHTML={{ __html: style.css }}
    />
  ));

  return {
    ...initialProps,
    emotionStyleTags,
  };
};
```

Is the custom _document the common theme?

**Comment by tomgreeEn** [comment]
> Also having this issue with 13.0.2
> 
> Production Builds using getStaticProps are failing on Vercel are failing with the error:
> 
> ```
> TypeError: Cannot read properties of undefined (reading 'getInitialProps')
> --
> 17:47:51.855 | at Object.41888 (/vercel/path0/.next/server/chunks/2843.js:3024:43)
> 17:47:51.855 | at __webpack_require__ (/vercel/path0/.next/server/webpack-runtime.js:25:42)
> 17:47:51.856 | at /vercel/path0/.next/server/chunks/2843.js:3294:94
> 17:47:51.856 | at Function.__webpack_require__.a (/vercel/path0/.next/server/webpack-runtime.js:89:13)
> 17:47:51.856 | at Object.34008 (/vercel/path0/.next/server/chunks/2843.js:3259:21)
> 17:47:51.856 | at __webpack_require__ (/vercel/path0/.next/server/webpack-runtime.js:25:42)
> 17:47:51.856 | at /vercel/path0/.next/server/chunks/2843.js:3219:93
> 17:47:51.856 | at Function.__webpack_require__.a (/vercel/path0/.next/server/webpack-runtime.js:89:13)
> 17:47:51.856 | at Object.6817 (/vercel/path0/.next/server/chunks/2843.js:3205:21)
> 17:47:51.856 | at __webpack_require__ (/vercel/path0/.next/server/webpack-runtime.js:25:42)
> ```
> 
> ...but I can do production builds locally.
> 
> swcMinify: false does not fix the error.
> 
> I am also using a custom _document to inject serverside styles, but using Emotion.
> 
> My _document file:
> 
> ```
> class MyDocument extends Document {
>   render() {
>     return (
>       <Html>
>         {...}
>       </Html>
>     );
>   }
> }
> 
> 
> MyDocument.getInitialProps = async (ctx) => {
>  
>   const originalRenderPage = ctx.renderPage;
> 
>   const cache = createEmotionCache();
>   const { extractCriticalToChunks } = createEmotionServer(cache);
> 
>   ctx.renderPage = () =>
>     originalRenderPage({
>       enhanceApp: (App) =>
>         function EnhanceApp(props) {
>           return <App emotionCache={cache} {...props} />;
>         },
>     });
> 
>   const initialProps = await Document.getInitialProps(ctx);
>   // This is important. It prevents Emotion rendering invalid HTML.
>   // See https://github.com/mui/material-ui/issues/26561#issuecomment-855286153
> 
>   const emotionStyles = extractCriticalToChunks(initialProps.html);
>   const emotionStyleTags = emotionStyles.styles.map((style) => (
>     <style
>       data-emotion={`${style.key} ${style.ids.join(' ')}`}
>       key={style.key}
>       // eslint-disable-next-line react/no-danger
>       dangerouslySetInnerHTML={{ __html: style.css }}
>     />
>   ));
> 
>   return {
>     ...initialProps,
>     emotionStyleTags,
>   };
> };
> ```
> 
> Is the custom _document the common theme?

UPDATE: Having disabled the part of _document with getInitialProps, and removed the other reference to getInitialProps in my app (inside the Sentry Error Handler), I still get this error.

**Comment by tomgreeEn** [comment]
> > Also having this issue with 13.0.2
> > Production Builds using getStaticProps are failing on Vercel are failing with the error:
> > ```
> > TypeError: Cannot read properties of undefined (reading 'getInitialProps')
> > --
> > 17:47:51.855 | at Object.41888 (/vercel/path0/.next/server/chunks/2843.js:3024:43)
> > 17:47:51.855 | at __webpack_require__ (/vercel/path0/.next/server/webpack-runtime.js:25:42)
> > 17:47:51.856 | at /vercel/path0/.next/server/chunks/2843.js:3294:94
> > 17:47:51.856 | at Function.__webpack_require__.a (/vercel/path0/.next/server/webpack-runtime.js:89:13)
> > 17:47:51.856 | at Object.34008 (/vercel/path0/.next/server/chunks/2843.js:3259:21)
> > 17:47:51.856 | at __webpack_require__ (/vercel/path0/.next/server/webpack-runtime.js:25:42)
> > 17:47:51.856 | at /vercel/path0/.next/server/chunks/2843.js:3219:93
> > 17:47:51.856 | at Function.__webpack_require__.a (/vercel/path0/.next/server/webpack-runtime.js:89:13)
> > 17:47:51.856 | at Object.6817 (/vercel/path0/.next/server/chunks/2843.js:3205:21)
> > 17:47:51.856 | at __webpack_require__ (/vercel/path0/.next/server/webpack-runtime.js:25:42)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ...but I can do production builds locally.
> > swcMinify: false does not fix the error.
> > I am also using a custom _document to inject serverside styles, but using Emotion.
> > My _document file:
> > ```
> > class MyDocument extends Document {
> >   render() {
> >     return (
> >       <Html>
> >         {...}
> >       </Html>
> >     );
> >   }
> > }
> > 
> > 
> > MyDocument.getInitialProps = async (ctx) => {
> >  
> >   const originalRenderPage = ctx.renderPage;
> > 
> >   const cache = createEmotionCache();
> >   const { extractCriticalToChunks } = createEmotionServer(cache);
> > 
> >   ctx.renderPage = () =>
> >     originalRenderPage({
> >       enhanceApp: (App) =>
> >         function EnhanceApp(props) {
> >           return <App emotionCache={cache} {...props} />;
> >         },
> >     });
> > 
> >   const initialProps = await Document.getInitialProps(ctx);
> >   // This is important. It prevents Emotion rendering invalid HTML.
> >   // See https://github.com/mui/material-ui/issues/26561#issuecomment-855286153
> > 
> >   const emotionStyles = extractCriticalToChunks(initialProps.html);
> >   const emotionStyleTags = emotionStyles.styles.map((style) => (
> >     <style
> >       data-emotion={`${style.key} ${style.ids.join(' ')}`}
> >       key={style.key}
> >       // eslint-disable-next-line react/no-danger
> >       dangerouslySetInnerHTML={{ __html: style.css }}
> >     />
> >   ));
> > 
> >   return {
> >     ...initialProps,
> >     emotionStyleTags,
> >   };
> > };
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Is the custom _document the common theme?
> 
> UPDATE: Having disabled the part of _document with getInitialProps, and removed the other reference to getInitialProps in my app (inside the Sentry Error Handler), I still get this error.

For me, the issue was @sentry/nextjs v 7.13+.
Downgrading to 7.12.1 has fixed it for now.

**Comment by karolina-szlenk** [comment]
@rossignolli I've noticed the same. Any progress? 
Many issues come from old browsers. Maybe a reason is that the old browsers do not support 'globalThis'? What do you think?
Can you reproduce it somehow, or only see it in Sentry issues?

**Comment by ShahriarDhruvo** [comment]
In v13 I get the same error after yarn build and yarn start. As I didn't use any getInitialProps in my project swcMinify=false solved this one for me... :(

**Comment by kdy1** [comment]
@ShahriarDhruvo Can you create a repro? Also, does it happen only for old browsers?

**Comment by jerocosio** [comment]
I was having a similar error after updating from Next.js 12 to Next.js 13, and the error only appeared on production, development worked great. 

This was the error I was getting:

`TypeError: Cannot read properties of undefined (reading 'getInitialProps') at Object.<anonymous>`

but adding `swcMinify: false,` to next.config.js fixed it

**Comment by kdy1** [comment]
Can someone provide a repro?

**Comment by DonikaV** [comment]
In my case error happens also only in production mode, and all browsers are up so date.
Can not show repo, sorry :(

**Comment by jerocosio** [comment]
@kdy1 Where can I send you a URL of my past deployments that had the issue? It's not a public project yet so I can't share it here.

**Comment by kdy1** [comment]
It does not help

**Comment by ShahriarDhruvo** [comment]
Sorry I changed a lot of things at once and not sure which one triggered it! As it reproduces when I deploy the project to the production and there are a lot of changes that goes with one version it is hard to find the exact reproduction case for me :(

I know you can't fix the actual bug if you can't reproduce it. I'll make some time to investigate into it later.

**Comment by JoBurg007** [comment]
I am faced with the same problem.
swcMinify: false did not solve it for me even in development. My Next version: 13.0.0, with react 18.2.0
![image](https://user-images.githubusercontent.com/53100131/201521438-a69b4f74-b525-4dc1-807c-78a793401716.png)

**Comment by kdy1** [comment]
@JoBurg007 Can you provide a repro?

**Comment by JoBurg007** [comment]
> @JoBurg007 Can you provide a repro?

@kdy1 Sorry, don't have this anymore. I meanwhile checked the web and implemented a solution that works for me now without making compromises and by using getInitialProps in _app without any error: https://jools.dev/server-side-auth-with-nextjs-context-and-hooks.
Again: I did not set swcMinify to false anymore. My Next version: 13.0.0, with react 18.2.0

**Comment by DonikaV** [comment]
I removed 
  swcMinify: false, 
  from my config and in new version of nextjs I don't have anymore this error.
  
```
  "loader-utils": "^3.2.1",
    "next": "^13.0.3",
    "next-transpile-modules": "^10.0.0",
```

**Comment by t-moennich** [comment]
FYI I also came across this error, straight after I added Sentry to my Next.js v12 project.
I didn't had a custom 404 page yet, but Sentry created a custom `_error.js`, which triggered the warning about a missing 404 page.
I quickly added one and got that error.
```
TypeError: Cannot read properties of undefined (reading 'getInitialProps') at Object.<anonymous>
```

After digging around a bit, I realized that I didn't had a `default export` in my `404.js` 🤦🏼 .
After changing the `export function Custom404()` to a `default export`, the error went away.

**tl;dr:** Verify, that you `404.js` (and of course all other pages` have the component as a `default export` 💡

**Comment by admmasters** [comment]
This is occurring for us with next 12.3.4:

Debugging has lead me to:

https://github.com/vercel/next.js/blob/4ca12bd5443cb124a97ab803a89b58cf4e8728b9/packages/next/shared/lib/utils.ts#L362

App is undefined and as such is unsafe to call `getInitialProps` on, maybe the check should assert that `App` is defined before calling `getInitialProps` on it?

**Comment by Leo1212** [comment]
> > > Also having this issue with 13.0.2
> > > Production Builds using getStaticProps are failing on Vercel are failing with the error:
> > > ```
> > > TypeError: Cannot read properties of undefined (reading 'getInitialProps')
> > > --
> > > 17:47:51.855 | at Object.41888 (/vercel/path0/.next/server/chunks/2843.js:3024:43)
> > > 17:47:51.855 | at __webpack_require__ (/vercel/path0/.next/server/webpack-runtime.js:25:42)
> > > 17:47:51.856 | at /vercel/path0/.next/server/chunks/2843.js:3294:94
> > > 17:47:51.856 | at Function.__webpack_require__.a (/vercel/path0/.next/server/webpack-runtime.js:89:13)
> > > 17:47:51.856 | at Object.34008 (/vercel/path0/.next/server/chunks/2843.js:3259:21)
> > > 17:47:51.856 | at __webpack_require__ (/vercel/path0/.next/server/webpack-runtime.js:25:42)
> > > 17:47:51.856 | at /vercel/path0/.next/server/chunks/2843.js:3219:93
> > > 17:47:51.856 | at Function.__webpack_require__.a (/vercel/path0/.next/server/webpack-runtime.js:89:13)
> > > 17:47:51.856 | at Object.6817 (/vercel/path0/.next/server/chunks/2843.js:3205:21)
> > > 17:47:51.856 | at __webpack_require__ (/vercel/path0/.next/server/webpack-runtime.js:25:42)
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > ...but I can do production builds locally.
> > > swcMinify: false does not fix the error.
> > > I am also using a custom _document to inject serverside styles, but using Emotion.
> > > My _document file:
> > > ```
> > > class MyDocument extends Document {
> > >   render() {
> > >     return (
> > >       <Html>
> > >         {...}
> > >       </Html>
> > >     );
> > >   }
> > > }
> > > 
> > > 
> > > MyDocument.getInitialProps = async (ctx) => {
> > >  
> > >   const originalRenderPage = ctx.renderPage;
> > > 
> > >   const cache = createEmotionCache();
> > >   const { extractCriticalToChunks } = createEmotionServer(cache);
> > > 
> > >   ctx.renderPage = () =>
> > >     originalRenderPage({
> > >       enhanceApp: (App) =>
> > >         function EnhanceApp(props) {
> > >           return <App emotionCache={cache} {...props} />;
> > >         },
> > >     });
> > > 
> > >   const initialProps = await Document.getInitialProps(ctx);
> > >   // This is important. It prevents Emotion rendering invalid HTML.
> > >   // See https://github.com/mui/material-ui/issues/26561#issuecomment-855286153
> > > 
> > >   const emotionStyles = extractCriticalToChunks(initialProps.html);
> > >   const emotionStyleTags = emotionStyles.styles.map((style) => (
> > >     <style
> > >       data-emotion={`${style.key} ${style.ids.join(' ')}`}
> > >       key={style.key}
> > >       // eslint-disable-next-line react/no-danger
> > >       dangerouslySetInnerHTML={{ __html: style.css }}
> > >     />
> > >   ));
> > > 
> > >   return {
> > >     ...initialProps,
> > >     emotionStyleTags,
> > >   };
> > > };
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Is the custom _document the common theme?
> > 
> > 
> > UPDATE: Having disabled the part of _document with getInitialProps, and removed the other reference to getInitialProps in my app (inside the Sentry Error Handler), I still get this error.
> 
> For me, the issue was @sentry/nextjs v 7.13+. Downgrading to 7.12.1 has fixed it for now.

Thanks, this solved my issue. I downgraded `@sentry/nextjs` to `7.12.1`
I got the following error:
```
TypeError: this.getNonce is not a function
    at getInitialProps (file:///pathToMyApp/node_modules/@rest-hooks/ssr/lib/nextjs/RestHooksDocument.js:29:19)
    at async getInitialProps (webpack-internal:///./pages/_document.page.tsx?__sentry_wrapped__:27:30)
    at async pathToMyApp\node_modules\@sentry\nextjs\cjs\config\wrappers\wrapperUtils.js:34:14
    at async pathToMyApp\node_modules\@sentry\nextjs\cjs\config\wrappers\wrapperUtils.js:113:14
```

**Comment by majid-amiri** [comment]
Hey everyone, any updates on this?

**Comment by pckilgore** [comment]
Can't confirm yet, but believe my case will be addressed by https://github.com/getsentry/sentry-javascript/pull/6572#event-8056853166

**Comment by FrancoAguzzi** [comment]
I was facing the same issue, tested the different fix approaches mentioned in this thread but none of them worked... So I started debugging branch commits one by one and found out the root cause of this error triggering in Cypress CI process was related to a code change I made to a index.ts file in the project. 

Basically, this index.ts file does the following inside the folder it is inserted:

export * from './component-1'
export * from './component-2'
export * from './component-3'

My error was updating the order of this exports, after undoing the order changes I made, the error was gone. 
Hope it helps someone going through de same.

**Comment by JClackett** [comment]
Still having this issue, seeing it sentry affect users using the below:

```

Chrome Mobile
Version:87.0.4280

Android
Version:8.0

Pixel 2
Model:Pixel 2
```

**Comment by Jay-flow** [comment]
It still occurs in the following environments.
- "@sentry/nextjs": "^7.34.0"
- "next": "13.2.4",

I tried upgrading to the latest version of @sentry/nextjs 7.53, but it didn't help much.

I don't think we've figured out exactly what the problem is yet.
Is there anyone who can clarify why this issue occurs and how to solve it?
If anyone leave a comment, it'll be a big help to everyone. Thanks.

**Comment by jt561** [comment]
Is anyone having the same issue with TRPC as a wrapper around their app?

**Comment by iscekic** [comment]
Any update? Seems like a simple `?.` as this comment says: https://github.com/vercel/next.js/issues/36221#issuecomment-1334450176

**Comment by jt561** [comment]
Turns out my issue was related to import statements. I am using a mono repo which had a lot of circular dependencies causing the build to have problems but everything works fine after creating a dependency graph and moving things around

**Comment by korac** [comment]
Leaving this comment for all those still experiencing the issue.

Same as stated in [this comment](https://github.com/vercel/next.js/issues/36221#issuecomment-1319229791), I started to experience the issue right after adding Sentry to my Next v12 project. 
The issue started to appear with `@sentry/nextjs: 7.17.1`.

[Changelog report](https://github.com/getsentry/sentry-javascript/blob/develop/CHANGELOG.md#7171) for `v7.17.1` lead me to [this PR](https://github.com/getsentry/sentry-javascript/pull/6013/) and specifically this line:
```ts
test: new RegExp(`^${escapeStringForRegex(pagesDirectory)}.*\\.(${pageExtensionRegex})$`),
```

This change was significant in my case since my project contained both `/pages` and `/pages-components` folders. Previously, given regex was matching only `/pages` folder but the new change included `/pages-components` as well.

Not completely familiar with how sentry boostraps in Next but renaming my `/pages-components` folder into something else (that doesn't start with `pages`) resolved the issue in my case.

ps. after this renaming, my projects is ok even with latest `@sentry/nextjs` version `7.57.0`.

**Comment by surbina** [comment]
> Not completely familiar with how sentry boostraps in Next but renaming my `/pages-components` folder into something else (that doesn't start with `pages`) resolved the issue in my case.

I also experienced this, but my project contained a `/pagesComponents` folder. Renaming the folder to something else did the trick. Thanks @korac.

**Comment by frattaro** [comment]
This started in one of our projects. I found a colleague had added (most likely auto-added on accident):

`import { NextRouter } from "next/router"`

removing that fixed it

**Comment by lsaavedr** [comment]
In my case the problem is fixed defining an `export default` in all .tsx files

**Comment by kevinrobert3** [comment]
@JClackett and @rossignolli do any of you have an update on this or a workaround? My issue affects ios devices on browser "Mobile Safari UI/WKWebView"

**Comment by kevinrobert3** [comment]
Turns out I needed just enough time and headache to troubleshoot this. 
For those that this affects iOS users, you need to debug on your phone (iOS), run local, and connect your phone browser.
All that same address and same network stuff. 
Then the error that usually occurs (in production) will be thrown, by nextjs. This will be easier for you to debug now since you even get the line it occurs from. 
For me, it was
1. A service worker for messaging that was not registering and throwing an error
2. A Firebase messaging error since I guess (firebase messaging) was not yet supported by iOS devices then and on old browsers

**Comment by samcx** [comment]
Hi everyone,

I will be moving this to <picture data-single-emoji=":nextjs:" title=":nextjs:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/nextjs/2a85d633ae594556.png" alt=":nextjs:" align="absmiddle"></picture> [Discussions](https://github.com/vercel/next.js/discussions) since there seems to be a variety of solutions that remedied this issue and a variety of issues that was/is causing this issue.

Thank you for all the different solutions and reports!

If the issue continues to arise, I encourage folks to create a separate [bug report](https://github.com/vercel/next.js/issues/new?assignees=&labels=template:+bug&projects=&template=1.bug_report.yml) for their issue. <picture data-single-emoji=":blob_bowing:" title=":blob_bowing:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/blob_bowing/3e13f8dbc67a5ea6.png" alt=":blob_bowing:" align="absmiddle"></picture>
